client/packages/system/src/Manage/Preferences/EditPage/EditPreference.tsx

```diff
@@ -30,10 +32,11 @@ export const EditPreference = ({
       }
       return (
         <Switch
+          disabled={disabled}
           checked={value}
           onChange={(_, checked) => {
             setValue(checked);
-            update({ [preference.key]: checked });
+            update(checked);
```

Comment:
Allow caller to handle the update as required (global prefs would just save the bool, store prefs we need to include the store id)